Library/ENV/4.3/cc

```diff
@@ -194,6 +196,27 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar)
+      %r[#{cellar}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(opt)
+      %r[#{opt}/([\w\-_\+]+)] =~ path
+      $1 && @deps.include?($1)
+    elsif path.start_with?(prefix)
+      true
+    else
+      !path.start_with?("/opt", "/sw", "/usr/X11")
```

Comment:
> In fact, if we are smart enough we could rewrite HOMEBREW_PREFIX path back to opt path if the path itself is a symlink created by Homebrew.

This seems like a very good idea to me 👍 

> I wonder will this break postgresql or its extensions https://github.com/Homebrew/homebrew/blob/master/Library/Formula/postgresql.rb#L44-L45

Like the sandbox I think it's maybe just worth figuring out how we want the behaviour to be and fixing up any cases that are broken.